File: ofos-client/src/features/address/services/addressService.js
import api from '../../../api/axiosConfig';

export const addressService = {
  // Get all addresses for current user
  getAllAddresses: () => {
    return api.get('/addresses');
  },

  // Get active addresses only
  getActiveAddresses: () => {
    return api.get('/addresses/active');
  },

  // Get default address
  getDefaultAddress: () => {
    return api.get('/addresses/default');
  },

  // Get address by ID
  getAddressById: (id) => {
    return api.get(`/addresses/${id}`);
  },

  // Add new address
  addAddress: (addressData) => {
    return api.post('/addresses', addressData);
  },

  // Update address
  updateAddress: (id, addressData) => {
    return api.put(`/addresses/${id}`, addressData);
  },

  // Set address as default
  setDefaultAddress: (id) => {
    return api.patch(`/addresses/${id}/default`);
  },

  // Delete address (soft delete)
  deleteAddress: (id) => {
    return api.delete(`/addresses/${id}`);
  },

  // Hard delete address
  hardDeleteAddress: (id) => {
    return api.delete(`/addresses/${id}/hard`);
  },

  // Get addresses by type
  getAddressesByType: (type) => {
    return api.get(`/addresses/type/${type}`);
  },
};

File: online-food-ordering-server/src/main/java/com/ofos/service/AddressService.java
package com.ofos.service;

import com.ofos.dto.request.AddressRequest;
import com.ofos.dto.request.AddressUpdateRequest;
import com.ofos.dto.response.AddressResponse;

import java.util.List;

public interface AddressService {
    
    AddressResponse addAddress(AddressRequest request, String userEmail);
    
    AddressResponse updateAddress(Long addressId, AddressUpdateRequest request, String userEmail);
    
    AddressResponse getAddressById(Long addressId, String userEmail);
    
    List<AddressResponse> getAllAddresses(String userEmail);
    
    List<AddressResponse> getActiveAddresses(String userEmail);
    
    List<AddressResponse> getAddressesByType(String addressType, String userEmail);
    
    AddressResponse getDefaultAddress(String userEmail);
    
    void setDefaultAddress(Long addressId, String userEmail);
    
    void deleteAddress(Long addressId, String userEmail);
    
    void softDeleteAddress(Long addressId, String userEmail);
}
